### PR TITLE
Add gitbase-spark-connector support

### PR DESCRIPTION
**Pull request by smacker**
- Use patched version of PyHive
- Update config script to produce correct connection string without name
and password (currently gsc requires connection without user/pass)

Signed-off-by: Maxim Sukharev <max@smacker.ru>

### CODE REVIEW

File: superset/requirements-dev.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ mysqlclient==1.3.13
 pip-tools==3.6.1
 psycopg2-binary==2.7.5
 pycodestyle==2.4.0
-pyhive==0.6.1
+-e git+https://github.com/src-d/PyHive.git@master#egg=PyHive
```
Comment (carlosms) [inline superset/requirements-dev.txt]: should we make a tag in our fork, instead of using master?

Comment (smacker) [reply superset/requirements-dev.txt]: DP team is responsible for that repo. So I don't know. I work with what I have.

Comment (smola) [reply superset/requirements-dev.txt]: cc @ajnavarro maybe we should tag something like `v0.6.1-srcd1`?

**Comment by smacker** [comment]
Just to be clear: this PR allows to initialize source{d} with gsc as a datasource. It doesn't mean source{d} works with gsc. (I couldn't execute any query, only get a list of tables)
But we will improve it later. (most probably with some help from DP team)

**Comment by smacker** [comment]
update: actually queries work. Only preview doesn't work because it selects from `schema.table` which doesn't work. Select from `table` works.

File: superset/requirements-dev.txt (hunk 1 above)
Comment (ajnavarro) [reply superset/requirements-dev.txt]: https://github.com/src-d/PyHive/releases/tag/v0.7.0-srcd1

**Pull request merged**